sky/serve/spot_policies.py

```diff
@@ -0,0 +1,133 @@
+""" Sky Spot Policy for SkyServe."""
+import collections
+import enum
+import typing
+from typing import Dict, List, Optional, Type
+
+from sky import sky_logging
+
+if typing.TYPE_CHECKING:
+    from sky.serve import replica_managers
+    from sky.serve import service_spec
+
+logger = sky_logging.init_logger(__name__)
+
+
+class Location:
```

Comment:
Could we use dataclass? IIRC it should have builtin __eq__ and __hash__

Reply:
I used it before but it doesn't work with dictionary.

Reply:
ic; then current impl LGTM